### PR TITLE
feat: add `updates-jenkins-io` service to `publick8s` cluster

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR adds an experimental `updates-jenkins-io` service on publick8s cluster

Ref: https://github.com/jenkins-infra/helpdesk/issues/2649#issuecomment-1660258021

### CODE REVIEW

File: clusters/publick8s.yaml [hunk 1]
```diff
@@ -234,3 +234,13 @@ releases:
...
     values:
       - "../config/ipv6-lb-service.yaml"
+  - name: uc-r2
+    namespace: uc-r2
+    chart: jenkins-infra/mirrorbits
+    version: 0.52.0
+    timeout: 600
+    atomic: false
+    values:
+      - "../config/uc-r2.yaml"
```
Comment (dduportal) [inline clusters/publick8s.yaml:244]:
```suggestion
      - "../config/updates.jenkins.io.yaml"
```

Nitpicking but:
- `updates.jenkins.io` is easier to understand than `uc` as the "Update Center" could mean different things
- R2 is not related to this installation: as we discussed together, R2 is for the controller mirrors

File: clusters/publick8s.yaml [hunk 2]
```diff
@@ -234,3 +234,13 @@ releases:
...
+    secrets:
+      - "../secrets/config/uc-r2/secrets.yaml"
```
Comment (dduportal) [inline clusters/publick8s.yaml:246]:
```suggestion
      - "../secrets/config/updates.jenkins.io/secrets.yaml"
```

same reason

File: config/uc-r2.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    pullPolicy: IfNotPresent
+
+ingress:
+  enabled: true
+  className: public-nginx
+  annotations:
+    "cert-manager.io/cluster-issuer": "letsencrypt-prod"
+    "nginx.ingress.kubernetes.io/ssl-redirect": "true"
+  hosts:
+    - host: test-hlm-uc-r2
```
Comment (dduportal) [inline config/uc-r2.yaml:18]:
```suggestion
    - host: azure.updates.jenkins.io
```

WDYT about using a valid hostname to allow testing the service as soon as it is installed?

File: config/uc-r2.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      paths:
+        - path: /
+          serviceNameSuffix: files
+        - path: /.*[.](deb|hpi|war|rpm|msi|pkg|sha256|md5sum|zip|gz|pdf|json|svg|sh|jpeg|ico|png|html)$
+  tls:
+    - secretName: uc-r2-tls
```
Comment (dduportal) [inline config/uc-r2.yaml:24]:
```suggestion
    - secretName: azure.updates.jenkins.io
```

If you want the TLS cert to be created, you need an hostname in a DNS zone in which Cert manager is set up to create DNS records

File: config/uc-r2.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      hosts:
+        - test-hlm-uc-r2.jenkins.io
```
Comment (dduportal) [inline config/uc-r2.yaml:26]:
```suggestion
        - azure.updates.jenkins.io
```

File: config/uc-r2.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          namespace: uc-r2
+      mountOptions:
+        - dir_mode=0755
+        - file_mode=0644
+        - uid=1000
+        - gid=1000
+        - mfsymlinks
+        - nobrl
+        - serverino
+        - cache=strict
```
Comment (dduportal) [inline config/uc-r2.yaml:83]: As discussed together: we should use an Azure Disk persistent volume here and ensure it is read-only on the "files" service (while read + write on the mirrorbits service)

File: config/uc-r2.yaml (hunk 3 above)
Comment (lemeurherve) [reply config/uc-r2.yaml]: That was the plan, I added the "test-hlm-uc-r2" CNAME record manually but forgot the `.jenkins.io` here.

(Record manually replaced by "azure.updates")

**Comment by lemeurherve** [comment]
Needs https://github.com/jenkins-infra/helm-charts/pull/580

**Comment by lemeurherve** [comment]
We'll have to wait a bit:

> Failed to wait for order resource "updates-jenkins-io-tls-8k49h-1994758939" to become r
> eady: order is in "errored" state: Failed to create Order: 429 urn:ietf:params:acme:error:rateLimited: Error creating new order :: too many certificates (5) already issued for t
> his exact set of domains in the last 168 hours: [azure.updates.jenkins.io](http://azure.updates.jenkins.io/), retry after 2023-08-09T00:55:28Z: see https://letsencrypt.org/docs/duplicate-certificate-limit/

**Comment by lemeurherve** [comment]
Blocked by https://github.com/jenkins-infra/helm-charts/pull/583

**Review by dduportal — APPROVED** [review]
LGTM, both code and helmfil diff looks good!

IIUC you already tried the installation manually (and cleaned it up) so go go go

**Pull request merged**